ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/SearchTree.java

```diff
@@ -65,9 +59,10 @@ void initLeaves(RaoData raoData) {
         previousDepthOptimalLeaf = rootLeaf;
     }
 
-    public CompletableFuture<RaoResult> run(RaoData raoData, RaoParameters raoParameters) {
+    public CompletableFuture<RaoResult> run(RaoData raoData, RaoParameters raoParameters, TreeParameters treeParameters) {
```

Comment:
I would make TreeParameters an extension of RaoParameters. Thus you would not need to add this argument to this method.

Reply:
we actually want to hide TreeParameters from the user. if we put it in RaoParameters, I think it might get confusing